File: 167-twoSumII/solution.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """

        # 85%
        d={};
        for i in range(len(numbers)):
            if target-numbers[i] in d:
                print numbers[i]
                return [d[target-numbers[i]]+1, i+1]
            else:
                d[numbers[i]] = i
        return []        


        head, tail = 0, len(numbers)


        # 25%
        head, tail = 0, len(numbers)-1
        
        while head<tail:
            if (numbers[tail]+numbers[head])>target:
                tail-=1
            elif (numbers[tail]+numbers[head])==target:
                return [head+1, tail+1]
            else:
                head+=1
                
        return []

File: 167-twoSumII/solution.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        // 30%
        Map<Integer, Integer> map = new HashMap();
        
        for(int i=0; i<numbers.length; i++){
            if(map.containsKey(numbers[i])){
                return new int[] {map.get(numbers[i])+1, i+1};
                
            }else{
                map.put(target-numbers[i], i);
            }
        }
        
        return new int[] {};


        // 100%
        int head=0, tail=numbers.length-1;
        while(head<tail){
            if((numbers[head]+numbers[tail])>target){
                tail = tail-1;
            }else if((numbers[head]+numbers[tail])<target){
                head = head+1;
            }else{
                return new int[] {head+1, tail+1};
            }
        }

        return new int[] {};



    }
}
